src/fabric.c

```diff
@@ -643,8 +690,32 @@ void fi_ini(void)
 			"Search for providers in specific path (default: "
 			PROVDLDIR ")");
 	fi_param_get_str(NULL, "provider_path", &provdir);
-	if (!provdir)
+	if (!provdir) {
 		provdir = PROVDLDIR;
+		
```

Comment:
The code below will skip handling for PROVDLDIR. @shefty - is it ok to handle LD_LIBRARY_PATH in absence of FI_RROVIDER_PATH? Or we still need to handle PROVDLDIR?

Reply:
My preference is:
If FI_PROVIDER_PATH has been specified, we *only* look in that location.
If FI_PROVIDER_PATH has not been set, we look in both the default provider path, as well as the standard library path.

This provides the best support for backwards compatibility.

Reply:
Is it correct understanding of high level logic?

provdir = getenv("FI_PROVIDER_PATH")
if (!provdir)
    provdir = PROVDLDIR + ":"; // to have empty dir at the end, i.e to handle system paths

dirs = ofi_split_and_alloc(provdir, ":")
for each dir in dirs
    ofi_ini_dir(dir)

And what happens if OFI registers providers with the same name - e.g. if we have them in PROVDLDIR and in LD)LIBRARY_PATH? Will the second one rewrite first one (if we consider both providers have the same version)?

Reply:
A DL provider will only replace one that is loaded if the version is newer.  Note that I added a change to the 1.10 release to improve provider versions, so that they actually change between releases.  :)

In your high-level logic, where is LD_LIBRARY_PATH being handled?

Reply:
>> In your high-level logic, where is LD_LIBRARY_PATH being handled?
If we make dlopen("lib<prov>.so") then it will be searched in LD_LIBRARY_PATH, correct? Although may be we also need to add "./" in the name.